examples/cpp/pyperf/PyPerfBPFProgram.cc

```diff
@@ -610,7 +603,11 @@ int read_python_stack(struct pt_regs* ctx) {
     }
 
     // read current PyFrameObject filename/name
-    event->stack[event->stack_len++] = read_symbol(state, cur_frame, cur_code_ptr);
+    // The compiler substitutes a constant for `i` because the loop is unrolled. This guarantees we
+    // are always within the array bounds. On the other hand, `stack_len` is a variable, so the
+    // verifier can't guarantee it's within bounds without an explicit check.
+    prog_stack[i] = read_symbol(state, cur_frame, cur_code_ptr);
+    event->stack_len++;
```

Comment:
This can be changed to `event->stack_len += i` at the end of this loop? Will save more insns

Reply:
i thought about this, but im not 100% sure about this because the loop has some `goto`s in the middle

Reply:
> Unless we can refer to the current value of i at the complete label which would be read from a stack variable that is updated with each iteration

No reason we should not be able to?

Reply:
> Is it worth it?

Nah, just noted. For the next micro optimziations run